lib/Service/HostedSignalingServerService.php

```diff
@@ -0,0 +1,535 @@
+					$this->logger->error('Getting the account information failed: account not found - HTTP status: ' . $status . ' Response body: ' . $body, ['app' => 'spreed']);
+
+					$message = $this->l10n->t('There is no such account registered.');
+					throw new HostedSignalingServerAPIException($message);
+				case Http::STATUS_INTERNAL_SERVER_ERROR:
+					$body = $response->getBody()->getContents();
+					$this->logger->error('Getting the account information failed: internal server error - HTTP status: ' . $status . ' Response body: ' . $body, ['app' => 'spreed']);
+
+					$message = $this->l10n->t('Something unexpected happened. Please try again later.');
+					throw new HostedSignalingServerAPIException($message);
+				default:
+					$body = $response->getBody()->getContents();
+					$this->logger->error('Getting the account information failed: something else happened - HTTP status: ' . $status . ' Response body: ' . $body, ['app' => 'spreed']);
+
+					$message = $this->l10n->t('Failed to fetch account information because the trial server behaved wrongly. Please check back later.');
+					throw new HostedSignalingServerAPIException($message);
+			}
+		} catch (\Exception $e) {
+			$this->logger->logException($e, [
+				'app' => 'spreed',
+				'message' => 'Failed to request hosted signaling server trial',
+			]);
+
+			$message = $this->l10n->t('Failed to fetch account information because the trial server is unreachable. Please check back later.');
+			throw new HostedSignalingServerAPIException($message);
+		}
+
+		$status = $response->getStatusCode();
+
+		if ($status !== Http::STATUS_OK) {
+			$body = $response->getBody();
+			$this->logger->error('Getting the account information failed: something else happened - HTTP status: ' . $status . ' Response body: ' . $body, ['app' => 'spreed']);
+
+
+			$message = $this->l10n->t('Something unexpected happened.');
+			throw new HostedSignalingServerAPIException($message);
+		}
+
+		$body = $response->getBody();
+		$data = json_decode($body, true);
+
+		if (json_last_error() !== JSON_ERROR_NONE) {
+			$this->logger->error('Getting the account information failed: cannot parse JSON response - JSON error: '. json_last_error() . ' ' . json_last_error_msg() . ' HTTP status: ' . $status . ' Response body: ' . $body, ['app' => 'spreed']);
+
+			$message = $this->l10n->t('Something unexpected happened.');
+			throw new HostedSignalingServerAPIException($message);
+		}
+
+		if (!isset($data['status'])
+			|| !isset($data['created'])
+			|| ($data['status'] === 'active' && (
+					!isset($data['signaling'])
+					|| !isset($data['signaling']['url'])
+					|| !isset($data['signaling']['secret'])
+				)
+			)
+			|| !isset($data['owner'])
+			|| !isset($data['owner']['url'])
+			|| !isset($data['owner']['name'])
+			|| !isset($data['owner']['email'])
+			|| !isset($data['owner']['language'])
+			|| !isset($data['owner']['country'])
+			/* TODO they are not yet returned
```

Comment:
@fancycode While doing a full test run I noticed that those fields are not yet returned. 

@nickvergessen Either leave it as it is now (the limits are only shown in the UI if present anyways) or coordinate with @fancycode if they are added.